src/textual/_compositor.py

```diff
@@ -386,33 +386,23 @@ def add_widget(
 
                 # Add any scrollbars
                 for chrome_widget, chrome_region in widget._arrange_scrollbars(
-                    container_size
+                    container_region
                 ):
                     map[chrome_widget] = MapGeometry(
-                        chrome_region + container_region.offset + layout_offset,
+                        chrome_region + layout_offset,
                         order,
                         clip,
                         container_size,
                         container_size,
                     )
 
-                if widget.is_container:
-                    # Add the container widget, which will render a background
-                    map[widget] = MapGeometry(
-                        region + layout_offset,
-                        order,
-                        clip,
-                        total_region.size,
-                        container_size,
-                    )
-                else:
-                    map[widget] = MapGeometry(
-                        child_region + layout_offset,
-                        order,
-                        clip,
-                        child_region.size,
-                        container_size,
-                    )
+                map[widget] = MapGeometry(
```

Comment:
These changes mean the compositor doesn't need to treat scrollable widgets differently, which is satisfying.